network.md

```diff
@@ -0,0 +1,65 @@
+
+### Networking Layers
+
+The design of the Revault network is logically separated into 4 layers:
+
+| Layer | Description |
+|-------|-------------|
+| Users | Defines users and their roles |
+| Revault | Handles application level semantics and authentication of messages passed across multiple machines|
+| Noise | Handles authenticated and encrypted channels between any two connected machines |
+| Transport | Handles TCP streams and Tor protocol for connections between any two machines |
+
+
+### Transport
+
+Machines communicate using TCP streams that are routed through Tor. 
+
+Clients will need to interact with the Tor protocol, but for practical reasons will not be expected to install and configure Tor. Instead a revault client will use https://github.com/MagicalBitcoin/libtor to handle connections with onion services. 
+
+All watchtowers, co-signing servers and the sync server will run as hidden/ onion services exposing a SOCKS5 proxy. The process for configuring Tor for servers that operate an onion service is defined at https://2019.www.torproject.org/docs/tor-onion-service.html.en.  
+
+> (TODO: automate this process for simple server management)
+
+### Noise Channels
+
+Authenticated and encrypted channels will be established using noise_KK or noise_KX. The notation X/K notation states whether the initiator's (first token) or the reponder's (second token) static public key is known (K) by the counterparty or is transmitted (X) during the handshake. So, for example, noise_KX means that the initiator's static public key is known to the responder before the handshake.
+
+Considering the flow of messages (https://github.com/re-vault/practical-revault/blob/master/messages.md) for spending and for routine transaction signing. We use the following definition of directionality for communications, with initiators on the left and responders on the right. 
+
+Stakeholder -> their Watchtower (Noise_KK)
+
+Stakeholder -> Sync Server (Noise_KX)
+
+Manager -> Cosigner (Noise_KX)
+
+Manager -> Sync Server (Noise_KX)  
+
+Watchtower -> Sync Server (Noise_KX)
+
+This determines which static public keys need to be generated during the initialization ceremony and with which entities they must be shared (as part of configuration information). 
+
+> TODO: write and link to ceremony.md
+
+We use noise channels with Curve25519 for Diffie-Hellman functions, ChaChaPoly for cipher functions and SHA256 for hash functions.
+
+### References
+
+https://noiseexplorer.com/patterns/KX/
+
+http://noiseprotocol.org/noise.html#application-responsibilities
```

Comment:
nit: Markdown allows you to link your references directly. Here is an example of usage: https://raw.githubusercontent.com/lightningd/plugins/master/README.md (see the bottom of the file)